rpc/gnmi/gnmi-specification.md

```diff
@@ -1247,19 +1247,27 @@ root +
 The path convention defined in [Section 2.2.2](#222-paths) allows nodes in the
 data tree to be identified by a unique set of node names (e.g.,`/a/b/c/d`) or
 paths that consist of node names coupled with attributes (e.g., `/a/e[key=10]`).
-In the case where where a node name plus attribute name is required to uniquely
-identify an element (i.e., the path within the schema represents a list, map, or
-array), the following considerations apply:
+In the case where a node name plus attribute name is required to uniquely
+identify an element (i.e., the path within the schema represents a list, map,
+or array), the following considerations apply:
 
 - In the case that multiple attribute values are required to uniquely address
-    an element - e.g., `/a/f[k1=10][k2=20]`- and a replace or update
-    operation's path specifies  a subset of the attributes (e.g., `/a/f[k1=10]`)
-    then this MUST be considered an error by the target system - and an status
-    code of`InvalidArgument (3)` specified.
+  an element - e.g., `/a/f[k1=10][k2=20]`- and a replace or update
+  operation's path specifies  a subset of the attributes (e.g., `/a/f[k1=10]`)
+  then this MUST be considered an error by the target system - and an status
+  code of` InvalidArgument (3)` specified.
+- In the case that key values are specified both as attributes of a node in the
+  path, and as their own elements within the value, then this MUST be considered
+  an error by the target system - and a status code of `InvalidArgument (3)`
+  specified.  Targeting a list member with update or replace operations that
+  modify it's own keys in either conflicting or identical fashion is not
```

Comment:
To support idempotent behavior, identical update/replace should be permitted.  Agreed that conflicting node / list key values must return an error.  I think that is already covered by the last bullet point in this stanza,